src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -8172,6 +8249,108 @@ instruct convF2HF_reg_reg(iRegINoSp dst, fRegF src, fRegF ftmp, iRegINoSp xtmp)
+%{
+  match(Set dst (ReinterpretS2HF src));
+  format %{ "fmv.h.x $dst, $src" %}
+  ins_encode %{
+    __ fmv_h_x($dst$$FloatRegister, $src$$Register);
+  %}
+  ins_pipe(fp_i2f);
+%}
+
+instruct convF2HFAndS2HF(fRegF dst, fRegF src)
+%{
+  match(Set dst (ReinterpretS2HF (ConvF2HF src)));
+  format %{ "convF2HFAndS2HF $dst, $src" %}
+  ins_encode %{
+    __ fcvt_h_s($dst$$FloatRegister, $src$$FloatRegister);
+  %}
+  ins_pipe(fp_uop_s);
+%}
+
+instruct reinterpretHF2S(iRegINoSp dst, fRegF src)
+%{
+  match(Set dst (ReinterpretHF2S src));
+  format %{ "fmv.x.h $dst, $src" %}
+  ins_encode %{
+    __ fmv_x_h($dst$$Register, $src$$FloatRegister);
+  %}
+  ins_pipe(fp_f2i);
+%}
+
+instruct convHF2SAndHF2F(fRegF dst, fRegF src)
+%{
+  match(Set dst (ConvHF2F (ReinterpretHF2S src)));
+  format %{ "convHF2SAndHF2F $dst, $src" %}
+  ins_encode %{
+    __ fcvt_s_h($dst$$FloatRegister, $src$$FloatRegister);
+  %}
+  ins_pipe(fp_uop_s);
+%}
+
+instruct sqrt_HF_reg(fRegF dst, fRegF src)
+%{
+  match(Set dst (SqrtHF src));
+  format %{ "fsqrt.h $dst, $src" %}
+  ins_encode %{
+    __ fsqrt_h($dst$$FloatRegister, $src$$FloatRegister);
+  %}
+  ins_pipe(fp_sqrt_s);
+%}
+
+instruct binOps_HF_reg(fRegF dst, fRegF src1, fRegF src2)
+%{
+  match(Set dst (AddHF src1 src2));
+  match(Set dst (SubHF src1 src2));
+  match(Set dst (MulHF src1 src2));
+  match(Set dst (DivHF src1 src2));
+  format %{ "binop_hf $dst, $src1, $src2" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+    switch(opcode) {
+      case Op_AddHF: __ fadd_h($dst$$FloatRegister, $src1$$FloatRegister, $src2$$FloatRegister); break;
+      case Op_SubHF: __ fsub_h($dst$$FloatRegister, $src1$$FloatRegister, $src2$$FloatRegister); break;
+      case Op_MulHF: __ fmul_h($dst$$FloatRegister, $src1$$FloatRegister, $src2$$FloatRegister); break;
+      case Op_DivHF: __ fdiv_h($dst$$FloatRegister, $src1$$FloatRegister, $src2$$FloatRegister); break;
+      default: assert(false, "%s is not supported here", NodeClassNames[opcode]); break;
+    }
+  %}
+  ins_pipe(fp_dop_reg_reg_s);
+%}
+
+instruct min_max_HF_reg(fRegF dst, fRegF src1, fRegF src2)
+%{
+  match(Set dst (MinHF src1 src2));
+  match(Set dst (MaxHF src1 src2));
+  format %{ "min_max_hf $dst, $src1, $src2" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+    switch(opcode) {
+      case Op_MinHF: __ minmax_fp($dst$$FloatRegister, $src1$$FloatRegister, $src2$$FloatRegister,
+                                  __ FLOAT_TYPE::half_precision, true);
+                     break;
+      case Op_MaxHF: __ minmax_fp($dst$$FloatRegister, $src1$$FloatRegister, $src2$$FloatRegister,
+                                  __ FLOAT_TYPE::half_precision, false);
+                     break;
+      default: assert(false, "%s is not supported here", NodeClassNames[opcode]); break;
+    }
+  %}
+  ins_pipe(pipe_class_default);
+%}
+
+instruct fma_HF_reg(fRegF dst, fRegF src1, fRegF src2)
+%{
+  match(Set dst (FmaHF src2 (Binary dst src1)));
```

Comment:
Question: Why is `dst` used as one of the source operands? There doesn't seem to be such a constraint at the instruction level. I didn't see similar constraint for F/D variants `maddF_reg_reg` and `maddD_reg_reg`. So it's likely to be relaxed.